api/src/services/authorization.js

```diff
@@ -452,47 +452,81 @@ const groupViewResultsById = (authorizationContext, viewResults) => {
   return _.groupBy(viewResults.rows, 'id');
 };
 
+const isTaskDoc = (row) => row.value && row.value.type === 'task';
+
 // casting everything to string to insure that comparisons are against the same type to avoid 5 > 'a' || 5 < 'a'
 const prepareForSortedSearch = array => array.map(element => String(element)).sort();
 const sortedIncludes = (sortedArray, element) => _.sortedIndexOf(sortedArray, String(element)) !== -1;
 
-const getAllowedDocIds = (feed, { includeTombstones = true } = {}) => {
-  return db.medic.query('medic/docs_by_replication_key', { keys: feed.subjectIds }).then(results => {
-    const validatedIds = [MEDIC_CLIENT_DDOC, getUserSettingsId(feed.userCtx.name)];
-    const tombstoneIds = [];
-    const viewResultsById = groupViewResultsById(feed, results);
+const getDocsByReplicationKey = (authorizationContext) => {
+  return db.medic.query('medic/docs_by_replication_key', { keys: authorizationContext.subjectIds }).then(results => {
+    const viewResultsById = groupViewResultsById(authorizationContext, results);
 
     // leverage binary search when looking up subjects
-    const sortedSubjects = prepareForSortedSearch(feed.subjectIds);
+    const sortedSubjects = prepareForSortedSearch(authorizationContext.subjectIds);
 
-    results.rows.forEach(row => {
+    const docsByReplicationKey = [];
+
+    results.rows.map(row => {
```

Comment:
```suggestion
    results.rows.forEach(row => {
```
Since the result isnt assigned